oci/default.nix

```diff
@@ -0,0 +1,556 @@
+{ lib
+, nix
+, closureInfo
+, runCommand
+, writeText
+, writeReferencesToFile
+, stdenv
+, python3
+}:
+
+let
+  inherit (builtins) head elem length toString;
+
+  readMetadata = item:
+    builtins.fromJSON (builtins.readFile "${item}/metadata.json");
+
+  makeAccounts = spec:
+    let
+      users = {
+        root = {
+          uid = 0;
+          group = "root";
+          home = "/root";
+          shell = "/bin/sh";
+        } // (spec.users.root or {});
+      } // (builtins.removeAttrs (spec.users or {}) [ "root" ]);
+
+      groups = {
+        root.gid = 0;
+      } // (spec.groups or {});
+
+      groupsForUser = { group, extraGroups ? [], ... }: [ group ] ++ extraGroups;
+
+      usersForGroup = group: builtins.concatStringsSep "," (
+        builtins.concatMap
+          (name: lib.optionals (elem group (groupsForUser users.${name})) [ name ])
+          (builtins.attrNames users)
+      );
+
+      makeGroupLine = name: { gid, ... }:
+        "${name}:x:${toString gid}:${usersForGroup name}";
+
+      makePasswdLine = name: { uid, home, group, shell, ... }:
+        "${name}:x:${toString uid}:${toString groups.${group}.gid}::${home}:${shell}";
+
+      makeHome = name: { uid, group, home, ... }: {
+        name = home;
+        value = {
+          type = "directory";
+          uid = uid;
+          gid = groups.${group}.gid;
+          mode = "0755";
+        };
+      };
+
+      concatLines = builtins.concatStringsSep "\n";
+    in
+
+    {
+      "etc/passwd" = {
```

Comment:
I'm not sure if `/etc/shadow` is needed. In my testing, stuff like `ls` and `whoami` seem to work just fine.